extensions/mssql/src/objectManagement/ui/databaseDialog.ts

```diff
@@ -431,6 +476,323 @@ export class DatabaseDialog extends ObjectManagementDialogBase<Database, Databas
 	}
 	//#endregion
 
+	//#region Database Properties - Data Scoped configurations Tab
+	private async initializeDatabaseScopedConfigurationSection(): Promise<void> {
+		this.dscOriginalData = deepClone(this.objectInfo.databaseScopedConfigurations);
+		const dscNameColumn: azdata.TableColumn = {
+			type: azdata.ColumnType.text,
+			value: localizedConstants.DatabaseScopedOptionsColumnHeader,
+			width: 220
+		};
+		const primaryValueColumn: azdata.TableColumn = {
+			type: azdata.ColumnType.text,
+			value: localizedConstants.ValueForPrimaryColumnHeader,
+			width: 105
+		};
+		const secondaryValueColumn: azdata.TableColumn = {
+			type: azdata.ColumnType.text,
+			value: localizedConstants.ValueForSecondaryColumnHeader,
+			width: 105
+		};
+
+		this.dscTable = this.modelView.modelBuilder.table().withProps({
+			columns: [dscNameColumn, primaryValueColumn, secondaryValueColumn],
+			data: this.objectInfo.databaseScopedConfigurations.map(metaData => {
+				return [metaData.name,
+				metaData.valueForPrimary,
+				metaData.valueForSecondary]
+			}),
+			height: getTableHeight(this.objectInfo.databaseScopedConfigurations.length, 1, DscTableRowLength),
+			width: DefaultTableWidth
+		}).component();
+
+		this.dscTabSectionsContainer.push(this.createGroup('', [this.dscTable], true));
+		this.disposables.push(
+			this.dscTable.onRowSelected(
+				async () => {
+					// When refreshing the data on primary/secondary updates, we dont need to validate/toggle inputs again
+					if (this.currentRowId !== this.dscTable.selectedRows[0]) {
+						this.currentRowId = this.dscTable.selectedRows[0];
+						await this.validateUpdateToggleDscPrimaryAndSecondaryOptions();
+					}
+
+					// This sets the focus to the primary/secondary inputs on data refresh
+					if (!isUndefinedOrNull(this.setFocusToInput)) {
+						await this.setFocusToInput.focus();
+					}
+				}
+			)
+		);
+	}
+
+	/**
+	 * Validating the selected database scoped configuration and updating the primary and secondary dropdown options and their selected values
+	 */
+	private async validateUpdateToggleDscPrimaryAndSecondaryOptions(): Promise<void> {
+		// Update the primary and secondary dropdown options based on the selected database scoped configuration
+		const isSecondaryCheckboxChecked = this.objectInfo.databaseScopedConfigurations[this.currentRowId].valueForPrimary === this.objectInfo.databaseScopedConfigurations[this.currentRowId].valueForSecondary;
```

Comment:
Since you're using `this.objectInfo.databaseScopedConfigurations[this.currentRowId]` in a lot of places here, caching that in a local variable at the to with a shorter name would help a lot with readability. The same goes for other parts of the file. Looks like there are 95 instances of that line in this file.

Reply:
yup, modified at required code parts.